src/DIRAC/Resources/Computing/BatchSystems/SLURM.py

```diff
@@ -79,7 +79,7 @@ def submitJob(self, **kwargs):
       if status != 0 or not output:
         break
 
-      lines = output.split('\n')
+      lines = str(output).split('\n')
```

Comment:
Why? I suspect you want to pass `universal_newlines=True` or `text=True` to `subprocess.Popen` instead?

Also applies to the others as well

Reply:
Because `output` and `error` here in py3 would be bytes.
These files need to be run also by py2 ("forever" as these are run with the python of the WN). 

`text=True` is only for py3.7+, and I missed I could use also `universal_newlines`. I will update tomorrow.

Reply:
> Because output and error here in py3 would be bytes.

In that case it currently does the wrong thing:

```python
>>> str(b"hello world")
"b'hello world'"
```

The "right" thing to do here is to do everything as `bytes` and prefix the "strings" like `b"\n"` though it's probably less work to add `universal_newlines=True` and hope the commands never spit out any invalid non-unicode sequences.